src/main/java/gregtech/integration/jei/recipe/GTRecipeWrapper.java

```diff
@@ -76,9 +79,26 @@ public void getIngredients(@Nonnull IIngredients ingredients) {
 
         // Outputs
         if (!recipe.getOutputs().isEmpty() || !recipe.getChancedOutputs().isEmpty()) {
-            List<ItemStack> recipeOutputs = recipe.getOutputs()
-                    .stream().map(ItemStack::copy).collect(Collectors.toList());
-            currentItemSlot += recipeOutputs.size();
+            List<ItemStack> recipeOutputs;
+
+            // Scanner Output replacing
+            if (recipe.getOutputs().get(0).hasTagCompound() && recipe.getOutputs().get(0).isItemEqual(MetaItems.TOOL_DATA_STICK.getStackForm(1))) {
+                ItemStack dataStick = recipe.getOutputs().get(0);
+                NBTTagCompound researchItemNBT = dataStick.getSubCompound("asslineOutput");
+                ItemStack researchItem = new ItemStack(researchItemNBT);
+                if (researchItem != null) {
+                    recipeOutputs = Collections.singletonList(researchItem);
+                } else {
+                    recipeOutputs = recipe.getOutputs()
+                            .stream().map(ItemStack::copy).collect(Collectors.toList());
+                    currentItemSlot += recipeOutputs.size();
```

Comment:
This code should be pulled out into a separate method so it does not need to be copy pasted and be in two locations.